### PR TITLE
Add base color palette

### LINKED ISSUE #48801: Implement New Color Tokens
https://www.figma.com/design/OlyKP09vro7EkQ2ZzXDDla/Colors?node-id=720-2712&amp;node-type=frame&amp;t=vP0BgaiN0moMhbNh-11

### PR DESCRIPTION
**Pull request by iethree**
closes https://github.com/metabase/metabase/issues/48801

### Description

Implements CSS Variables for our entire new color palette. Also implements a lint rule prohibiting reference to these colors in the application. (doesn't prohibit their use in CSS files yet)

The next step is to map the base colors onto semantic colors and see how things look;

Screen Shot 2024-10-30 at 1 16 21 PM

### CODE REVIEW

File: frontend/src/metabase/css/core/colors.module.css [hunk 1]
```diff
@@ -158,17 +154,132 @@
...
+  --mb-base-color-flamingo-90: hsla(334, 79%, 17%, 1);
+  --mb-base-color-flamingo-80: hsla(334, 72%, 26%, 1);
+  --mb-base-color-flamingo-70: hsla(334, 71%, 36%, 1);
+  --mb-base-color-flamingo-60: hsla(334, 69%, 48%, 1);
+  --mb-base-color-flamingo-50: hsla(334, 67%, 60%, 1);
+  --mb-base-color-flamingo-40: hsla(334, 80%, 68%, 1);
+  --mb-base-color-flamingo-30: hsla(334, 79%, 80%, 1);
+  --mb-base-color-flamingo-20: hsla(335, 79%, 91%, 1);
+  --mb-base-color-flamingo-10: hsla(335, 67%, 96%, 1);
+  --mb-base-color-flamingo-5: hsla(330, 67%, 99%, 1) /* Mango */;
```
Comment (kdoh) [inline frontend/src/metabase/css/core/colors.module.css:219]: Looks like the formatting got wonky here.

File: frontend/src/metabase/css/core/colors.module.css [hunk 2]
```diff
@@ -1,9 +1,24 @@
...
+ *
+ * 1. Semantic colors
+ * 2. Base colors
+ *
+ * Semantic colors are the colors that are used in the application. They are derived
+ * from the base colors.
+ *
+ * Base colors are the colors that should never change and never be used outside this file,
+ * or in any components or CSS properties. They should only be used to define the semantic
+ * colors in this file.
```
Comment (WiNloSt) [inline frontend/src/metabase/css/core/colors.module.css:19]: I think there are still cases for base colors to be defined outside this file.
1. ad-hoc colors: Like colors that are one-off, we could potentially create one-off semantic colors that are based on base colors e.g. I did this for [SVG icons](https://github.com/metabase/metabase/blob/64f130cc7f090918c20c0d242d7d8660b9c1f0c7/frontend/src/metabase/admin/settings/components/widgets/EmbeddingOption/EmbeddingOption.module.css) with different color schemes
2. Theming: We want to be able to override these semantic colors elsewhere, for example
    https://github.com/metabase/metabase/blob/17cefb94c533492a7e6bb9fef312d222798ea24a/frontend/src/metabase/public/components/EmbedFrame/EmbedFrame.module.css#L59-L68
    Or do you think such overrides should live within this file?

Comment (iethree) [reply frontend/src/metabase/css/core/colors.module.css:19]: 🤔

File: frontend/src/metabase/admin/settings/components/widgets/EmbeddingOption/use-embedding-settings-icon-colors.tsx [hunk 3]
```diff
@@ -2,6 +2,7 @@ export type UseEmbeddingSettingsIconColorsProps = {
   disabled: boolean;
 };
 
+/* eslint-disable no-restricted-syntax -- we should find a way to use semantic colors here */
```
[the remark this answers is not in the document]
Comment (oisincoveney) [reply frontend/src/metabase/admin/settings/components/widgets/EmbeddingOption/use-embedding-settings-icon-colors.tsx:5]: taking a look right now

[the remark this answers is not in the document]
Comment (oisincoveney) [reply frontend/src/metabase/admin/settings/components/widgets/EmbeddingOption/use-embedding-settings-icon-colors.tsx:5]: https://github.com/metabase/metabase/pull/49478

[the remark this answers is not in the document]
Comment (oisincoveney) [reply frontend/src/metabase/admin/settings/components/widgets/EmbeddingOption/use-embedding-settings-icon-colors.tsx:5]: Let me know what you think - I also noticed that the linter isn't hitting the CSS files when I rebased my branch on top of this one

File: frontend/src/metabase/css/core/colors.module.css (hunk 2 above)
Comment (WiNloSt) [reply frontend/src/metabase/css/core/colors.module.css:19]: resolving. I also [agree that we should declare all semantic colors inside `colors.module.css`](https://metaboat.slack.com/archives/C057WD5L0JG/p1730732847564649?thread_ts=1730317945.468619&cid=C057WD5L0JG)

**Comment by oisincoveney** [comment]
Merging master to add the `custom-column` flake fix

**Comment by iethree** [comment]
> Merging master to add the `custom-column` flake fix

pro tip: rebasing uses less CI resources,  merging pulls in all CI changes.

**Review by WiNloSt — APPROVED** [review]
Looks good to me.

**Review by oisincoveney — APPROVED** [review]
i like!

**Pull request merged**